src/pages/index.js

```diff
@@ -48,11 +133,42 @@ const profilePopup = new PopupWithForm(
   handleProfileFormSubmit
 );
 
-const user = new UserInfo(".profile__title", ".profile__description");
-
 function handleProfileFormSubmit({ title, description }) {
-  user.setUserInfo(title, description);
-  profilePopup.close();
+  renderSaving(editProfileModalSelector, true);
+  api
+    .editProfile("users", "me", title, description)
+    .then((response) => {
+      user.setUserInfo(title, description);
+      profilePopup.close();
+    })
+    .catch((error) => {
+      console.log(error);
+    })
+    .finally(() => {
+      renderSaving(editProfileModalSelector, false);
+    });
+}
+
+function renderSaving(popupSelector, isLoading = false) {
+  const saveButtonEl = document.querySelector(
+    `${popupSelector} .modal__button`
+  );
+  if (isLoading) {
+    saveButtonEl.textContent = "Saving...";
+  } else {
+    saveButtonEl.textContent = "Save";
+  }
+}
+
+function renderCreating(popupSelector, isLoading = false) {
```

Comment:
tHe code for rendering loading gets duplicated. You can either add arguments `buttonText` and `loadingButtonText` to one of the loading methods or implement the loading via class methods like in the comment in FormValidator class